cpp/src/plasma/protocol.cc

```diff
@@ -230,35 +230,63 @@ Status ReadReleaseReply(uint8_t* data, size_t size, ObjectID* object_id) {
   return plasma_error_status(message->error());
 }
 
-// Delete messages.
+// Delete objects messages.
 
-Status SendDeleteRequest(int sock, ObjectID object_id) {
+Status SendDeleteRequest(int sock, const std::vector<ObjectID>& object_ids) {
   flatbuffers::FlatBufferBuilder fbb;
-  auto message = CreatePlasmaDeleteRequest(fbb, fbb.CreateString(object_id.binary()));
+  auto message =
+      CreatePlasmaDeleteRequest(fbb, static_cast<int>(object_ids.size()),
+                                to_flatbuffer(&fbb, &object_ids[0], object_ids.size()));
   return PlasmaSend(sock, MessageType::PlasmaDeleteRequest, &fbb, message);
 }
 
-Status ReadDeleteRequest(uint8_t* data, size_t size, ObjectID* object_id) {
+Status ReadDeleteRequest(uint8_t* data, size_t size, std::vector<ObjectID>& object_ids) {
```

Comment:
The google styleguide here is that out arguments must be pointers (and all references must be const): https://google.github.io/styleguide/cppguide.html#Reference_Arguments

This is a matter of convention of course, but in arrow we have been adhering to it and so should continue to do so.